Compiler Error: Comparison of integer expressions of different signedness

Hi,
I wanted to upgrade to the latest 0.4.1 version.
I installed the prebuilt `au_all_units.hh` and get following compile error:

```console
../source/Common/Units/include/Units/au.h: In function ‘constexpr uintmax_t au::detail::find_prime_factor(uintmax_t)’:

[2025-04-30T11:22:07.560Z] ../source/Common/Units/include/Units/au.h:2883:19: error: comparison of integer expressions of different signedness: ‘int’ and ‘uintmax_t’ {aka ‘long unsigned int’} [-Werror=sign-compare]

[2025-04-30T11:22:07.560Z]  2883 |         if (p * p > n) {
[2025-04-30T11:22:07.560Z]       |             ~~~~~~^~~
```

This is a snippet of `constexpr std::uintmax_t find_prime_factor(std::uintmax_t n)`

It fails with different GCC versions like 9.3 and 13.2

My sub-issue is to enable warnings as errors in your CI: -Wall -Wextra -Werror

--- a/au/code/au/utility/factoring.hh
+++ b/au/code/au/utility/factoring.hh
@@ -96,7 +96,7 @@ constexpr std::uintmax_t find_prime_factor(std::uintmax_t n) {
     //
     // Note that range-for isn't supported until C++17, so we need to use an index.
     for (auto i = 0u; i < FirstPrimes::values.size(); ++i) {
-        const auto &p = FirstPrimes::values[i];
+        const std::uintmax_t p = FirstPrimes::values[i];
 
         if (n % p == 0u) {
             return p;

